W 12985329 jenkins cli pipeline gr (#253)
mulesoft/docs-gateway: gateway/1.4/modules/ROOT/pages/flex-conn-jenkins-api.adoc

ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]
:imagesdir: ../assets/images
:page-mode: conn
:page-aliases: gateway::flex-conn-manage-jenkins.adoc

You can use a Jenkins Pipeline to automate the workflow for Flex Gateway in Connected Mode. With a Jenkins Pipeline, you can register and run a Flex Gateway and use features of the API Manager API, such as creating and deploying API instances.

Automating your Flex Gateway in Connected Mode work flow shares many similarities with the xref:flex-conn-manage-public-api.adoc[Manage Flex Gateway Using the API Manager API] tutorial. You can use the Jenkins Pipeline to automate the calls made to the API Manager API. To add additional features from the  https://anypoint.mulesoft.com/exchange/portals/anypoint-platform/f1e97bc6-315a-4490-82a7-23abe036327a.anypoint-platform/api-manager-api/[API Manager API^], use the <<create-an-api-instance, Create an API Instance>> and <<apply-an-api-policy, Apply an API Policy>> Pipeline stages as examples.

This tutorial has different steps with different code blocks. As you work through the steps, add the code blocks to the same Jenkinsfile. 

> NOTE: If you prefer to automate your Flex Gateway with the Anypoint CLI, see xref:flex-conn-jenkins-cli.adoc[].

Reviewer: ```suggestion
NOTE: To instead automate your Flex Gateway with the Anypoint CLI, see xref:flex-conn-jenkins-cli.adoc[].
```

= Automating Flex Gateway with a Jenkins Pipeline using the API Manager API
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]
:imagesdir: ../assets/images
:page-mode: conn

> :page-aliases: flex-conn-manage-jenkins.adoc

Reviewer 2: ```suggestion
:page-aliases: gateway::flex-conn-manage-jenkins.adoc
```
Just to be safe (FYI the component is only needed in page aliases, not xrefs to pages within the same component)
